app/scripts/models/marketing-email-prefs.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'backbone',
+  'lib/promise',
+  'lib/constants'
+], function (Backbone, p, Constants) {
+
+  var NEWSLETTER_ID = Constants.MARKETING_EMAIL_NEWSLETTER_ID;
+  var SCOPES = Constants.MARKETING_EMAIL_OAUTH_SCOPE;
+  var MANAGEMENT_BASE_URL = Constants.MARKETING_EMAIL_MANAGEMENT_BASE_URL;
+
+  var MarketingEmailPrefs = Backbone.Model.extend({
+    defaults: {
+      managementUrl: null,
+      isOptedIn: null
+    },
+
+    initialize: function (options) {
+      options = options || {};
+
+      var self = this;
+      self._marketingEmailClient = options.marketingEmailClient;
+      self._account = options.account;
+    },
+
+    destroy: function () {
+      var self = this;
+      if (self._accessToken) {
+        return self._account.destroyOAuthToken(self._accessToken);
```

Comment:
Something about calling the account to destroy the oauth token doesn't seem right.